src/pieces/logs/serverLog.ts

```diff
@@ -309,6 +310,41 @@ async function processMessageDelete(msg: Message | PartialMessage, serverLog: Te
 	}
 
 	serverLog.send({ embeds: [embed], files: attachments });
+
+	const bot = msg.client;
+	handleExpDetract(bot, msg);
+}
+
+async function handleExpDetract(bot: Client, msg: Message | PartialMessage) {
+	const user: SageUser = await msg.author.client.mongo.collection(DB.USERS).findOne({ discordId: msg.author.id });
+	console.log(user);
+	// bot.mongo.collection(DB.USERS).findOneAndUpdate(
+	// 	{ user },
+	// 	{ $inc: { count: -1, curExp: +1 } }
+	// );
+	if (user.curExp < user.levelExp) {
+		bot.mongo.collection(DB.USERS).findOneAndUpdate(
+			{ discordId: msg.author.id },
+			{ $inc: { count: 0, curExp: +1 } }
+		);
+	} else { // we can't have negative exp
+		bot.mongo.collection(DB.USERS).findOneAndUpdate(
+			{ discordId: msg.author.id },
+			{ $inc: { count: 0, curExp: 0 } }
+		);
+	}
+
+	if (user.count < 1) {
+		bot.mongo.collection(DB.USERS).findOneAndUpdate(
+			{ discordId: msg.author.id },
+			{ $inc: { count: 0, curExp: 0 } }
+		);
+	} else { // we can't have negative message counts
+		bot.mongo.collection(DB.USERS).findOneAndUpdate(
+			{ discordId: msg.author.id },
+			{ $inc: { count: -1, curExp: 0 } }
+		);
+	}
```

Comment:
Afaik, you can assign more than one handler to a single event. I think this code would be better suited in the [message count piece.](https://github.com/ud-cis-discord/SageV2/blob/main/src/pieces/messageCount.ts*)